beacon-chain/blockchain/process_block.go

```diff
@@ -330,7 +330,15 @@ func (s *Service) handleBlockAfterBatchVerify(ctx context.Context, signed *ethpb
 
 // Epoch boundary bookkeeping such as logging epoch summaries.
 func (s *Service) handleEpochBoundary(ctx context.Context, postState *stateTrie.BeaconState) error {
-	if postState.Slot() >= s.nextEpochBoundarySlot {
+	if postState.Slot() == s.nextEpochBoundarySlot-1 {
```

Comment:
any reason we changed from your previous approach here ?

Reply:
1.) We should only `reportEpochMetrics` at epoch boundary slot

2.) If for whatever reason epoch boundary slot - 1 is a skip slot, we should: 
`helpers.UpdateCommitteeCache(postState, helpers.CurrentEpoch(postState))` 
instead of:
`helpers.UpdateCommitteeCache(postState, helpers.NextEpoch(postState))`